src/index.ts

```diff
@@ -1642,7 +1673,12 @@ export type OrderBy =
 	| string
 	| OrderBy[]
 	| Dictionary<OrderByDirection>
-	| (Dictionary<{ $count: ODataCountOptions }> & { $dir: OrderByDirection });
+	| {
+			[k: StartsWithLetter]: {
```

Comment:
Unfortunately TS atm doesn't support negated types (ie `Exclude<string, '$dir'> => string`), so this was the best I found working.
Also tried conditional types but they still evaluated to the same expression.